rollup/internal/controller/relayer/l1_relayer.go

```diff
@@ -230,6 +234,20 @@ func (r *Layer1Relayer) StopSenders() {
 }
 
 func (r *Layer1Relayer) shouldUpdateGasOracle(baseFee uint64, blobBaseFee uint64, isCurie bool) bool {
+	fields := map[string]interface{}{
+		"rollup_status": types.RollupCommitting,
+	}
+	orderByList := []string{"updated_at DESC"}
+	limit := 1
+	batches, err := r.batchOrm.GetBatches(r.ctx, fields, orderByList, limit)
+	if err != nil {
+		log.Error("failed to fetch latest committing batches", "err", err)
+	}
+	// do not update GasOracle if we haven't committed batches for a while
+	if len(batches) == 0 || utils.NowUTC().Sub(batches[0].UpdatedAt) > time.Duration(r.cfg.GasOracleConfig.GasPriceOracleUpdateWindow)*time.Minute {
```

Comment:
We should remove `len(batches) == 0`, we'd also match that if we successfully committed all pending batches.